src/main/java/com/conveyal/datatools/manager/controllers/api/DeploymentController.java

```diff
@@ -438,19 +459,17 @@ private static String deploy (Request req, Response res) {
             );
         }
 
-        // For any previous deployments sent to the server/router combination, set deployedTo to null because
-        // this new one will overwrite it. NOTE: deployedTo for the current deployment will only be updated after the
-        // successful completion of the deploy job.
-        for (Deployment oldDeployment : Deployment.retrieveDeploymentForServerAndRouterId(target, deployment.routerId)) {
-            LOG.info("Setting deployment target to null id={}", oldDeployment.id);
-            Persistence.deployments.updateField(oldDeployment.id, "deployedTo", null);
-        }
-
         // Execute the deployment job and keep track of it in the jobs for server map.
         DeployJob job = new DeployJob(deployment, userProfile, otpServer);
-        DataManager.heavyExecutor.execute(job);
-        deploymentJobsByServer.put(target, job);
-
+        if (!queueDeployJob(job)) {
+            // Job for the target is still active! Send a 202 to the requester to indicate that it is not possible
+            // to deploy to this target right now because someone else is deploying.
+            String message = String.format(
+                "Will not process request to deploy %s. Deployment currently in progress for target: %s",
```

Comment:
I have commented on this before, it looks like this error log entry and the error log entry from `queueDeployJob` are redundant. Perhaps remove the log generated by `queueDeployJob`?

Reply:
Leaving as is to cover future use where a false response might not be logged.